modules/search/reddit.py

```diff
@@ -31,71 +31,69 @@ class Module(BaseModule):
 			('limit', 1, False, 'Search limit(number of pages, default=1)', '-l', 'store'),
 			('count', 50, False, 'Number of links per page(min=10, max=100, default=50)', '-c', 'store'),
 			('engine', 'google', False, 'Engine names for search(default=google)', '-e', 'store'),
+			('thread', 2, False, 'The number of engine that run per round(default=2)', '-t', 'store'),
 			('output', False, False, 'Save output to workspace', '--output', 'store_true'),
 		),
 		'examples': ('reddit -q <QUERY> -l 15 --output',)
 	}
 
+	links = []
+
+	def set_data(self, urls):
+		for url in urls:
+			self.links.append(url)
+
+	def thread(self, function, thread_count, engines, q, q_format, limit, count):
+		threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=thread_count)
+		futures = (threadpool.submit(
+			function, name, q, q_format, limit, count) for name in engines if name in self.meta['sources'])
+		for _ in concurrent.futures.as_completed(futures):
+			pass
+
+	def search(self, name, q, q_format, limit, count):
+		engine = getattr(self, name)
+		name = engine.__name__
+		query = name + '_q'
```

Comment:
don't use try/except for this such trivial tasks,
`query = q_formats[f"{name}_q"] if f"{name}_q" in q_formats else q_formats['default_q']`
q_formats variable name also is better